tests/systemtest/test_content_length.py

```diff
@@ -20,7 +20,8 @@ def http_post(posturl, headers, data):
     if ':' in parsed.netloc:
         host, port = parsed.netloc.split(':')
     else:
-        host, port = parsed.netloc, '80'
+        host = parsed.netloc
+        port = '443' if parsed.startswith('https') else '80'
```

Comment:
I think you either want `parsed.scheme == https` or `posturl.startswith('https')`.